components/tabs/src/sync/record.rs

```diff
@@ -27,7 +28,163 @@ pub struct TabsRecord {
 impl TabsRecord {
     #[inline]
     pub fn from_payload(payload: sync15::Payload) -> Result<Self> {
-        let record: TabsRecord = payload.into_record()?;
-        Ok(record)
+        let client_name: String = payload.data["clientName"]
+            .as_str()
+            .unwrap_or_default()
```

Comment:
Hrm, so I think there's a subtle change in behaviour here. If `clientName` is present but is not a string, IIUC the previous code would have given an error, while the new code will default it to the empty string. I don't have a super strong opinion about which is the better behaviour, but it may be useful to opt for consistency with other engines and continue to error out in this case. @lougeniaC64 WDYT?

(If we do want to error out here, [`Option::ok_or`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or) may be helpful, and we should add some tests to cover missing or badly-typed data)